pkg/scheduling/v1/queuer.go

```diff
@@ -407,6 +407,11 @@ func (q *Queuer) ack(r *assignResults) {
 		delete(q.unassigned, schedulingTimedOutItem.ID)
 	}
 
+	for _, rateLimitedItemToMove := range r.rateLimitedToMove {
+		delete(q.unacked, rateLimitedItemToMove.qi.ID)
+		delete(q.unassigned, rateLimitedItemToMove.qi.ID)
+	}
```

Comment:
This is the crucial fix - we needed delete the items from `q.unassigned` so they aren't just read in again on requeue